make-manifest.sh

```diff
@@ -0,0 +1,269 @@
+      local tag="${mirror_repo}:${version}-${arch}"
+      local annotate_args="--os ${os} --arch ${arch}"
+      local platform="${os}/${arch}"
+      if [ -n "${variant}" ]; then
+        tag+="-${variant}"
+        annotate_args+=" --variant ${variant}"
+        platform+="/${variant}"
+      fi
+
+      local img_name=""
+      if [ -n "${image_host}" ]; then
+	  img_name="${image_host}/"
+      fi
+      img_name+="${arch_img}"
+
+      tag-arch-img() {
+	if [ $(check_if_img_with_tag_already_exists ${tag}) -ne 0 ]; then
+	    info "... Pulling image from source repo: ${img_name}"
+            docker pull -q ${img_name} >/dev/null
+	    info "... Tagging original source repo image: ${tag}"
+            docker tag ${img_name} ${tag} >/dev/null
+	else
+	    info "... Pulling preexisting image from destination repo with tags: ${tag}"
+	    docker pull -q ${tag} >/dev/null
+	fi
+      }
+
+      set-image-platform() {
+        info "... Set platform ${platform} for image ${tag}"
+        mkdir -p ${tag}
+        docker image save ${tag} -o ${tag}.tar >/dev/null
+        for f in $(tar --list -f ${tag}.tar | grep -e '[./]json$'); do
+          tar -C ${tag} -xf ${tag}.tar ${f}
+          if jq '
+              if has("os") then .os = "'${os}'" else . end |
+              if has("architecture") then .architecture = "'${arch}'" else . end |
+              if has("variant") then .variant = "'${variant}'" else . end
+            ' <${tag}/${f} >${tag}/${f}.tmp 2>/dev/null; then
+            mv -f ${tag}/${f}.tmp ${tag}/${f}
+            tar -C ${tag} -uf ${tag}.tar ${f}
+          fi
+        done
+        docker image load -q -i ${tag}.tar >/dev/null
+        rm -rf ${tag}*
+      }
+
+      annotate-manifest() {
+	if [ $(check_if_img_with_tag_already_exists ${tag}) -ne 0 ]; then
+	    info "... Pushing new image to destination repo with tags: ${tag}"      
+	    docker push ${tag} >/dev/null	
+	fi
+        local digest=$(docker image inspect ${tag} | jq -r '.[] | .RepoDigests[0]')
+	info "... Creating manifest: amend-option:${manifest_amend} manifest-name:${mirror} image:${digest}"
+        docker manifest create ${manifest_amend} ${mirror} ${digest} >/dev/null
+        docker manifest annotate ${mirror} ${digest} ${annotate_args} >/dev/null
+        docker image rm -f ${tag} >/dev/null
+        manifest_amend='--amend'
+      }
+
+      {
+        tag-arch-img
+        set-image-platform
+        annotate-manifest
+      }
+
+    done < <(manifest-list ${image})
+    info "--- Push mirror ${mirror}"
+    docker manifest push ${mirror} >/dev/null
+  }
+
+  lock "${mirror_lock}"
+  (
+    trap 'cleanup-mirror ${mirror} ${mirror_lock}' EXIT
+    info "+++ Create mirror ${mirror}"
+    process-manifest    
+  )
+}
+
+manifest-inspect() {
+  # docker manifest inspect $1 | \
+  #     jq -r '.manifests[] | "'$1'@\(.digest) \(.platform.os) \(.platform.architecture) \(.platform.variant // "")"'
+  manifest-tool inspect --raw $1 2>/dev/null | \
```

Comment:
check if `docker manifest inspect` can do the same thing and we can remove manifest-tool as a dependency.

Reply:
docker manifest inspect only seems to be working on dockerhub images and throws error for other repos.